src/response/response_validation.rs

```
@@ -45,8 +45,13 @@ pub fn validation_response_full(
         .operation_id
         .clone()
         .unwrap_or("operationWithNoId".to_string());
+        
+    let status_code: StatusValue = match operation.responses.iter().find(|kv| kv.0.parse::<i64>().unwrap() < 400) {
+        Some(r) => StatusValue::Specific(r.0.parse::<u64>().unwrap()),
+        None => return Ok(Some(validate_response_not_error())),
+    };
 
-    let response = match operation.responses.iter().find(|kv| kv.0 == "200") {
+    let response = match operation.responses.iter().find(|kv| kv.0.parse::<i64>().unwrap() < 400) {
```

This is a really good change, thanks for this.

I'm concerned this unwrap will panic though. But I like this, it should have worked more like this from the beginning.

What do you think about parsing just the first character (1,2,3,4,5) and then matching on the error to see if the value is "default"?

If the first character is not a number and the word is not "default" then the oai spec is invalid, but I'd still prefer not to panic. We could return an error in that case, but it would be more convenient to default to just ignore the value in the filter.